src/main/kotlin/io/emeraldpay/dshackle/upstream/grpc/EthereumPosGrpcUpstream.kt

```diff
@@ -125,10 +125,13 @@ open class EthereumPosGrpcUpstream(
     override fun stop() {
     }
 
-    override fun update(conf: BlockchainOuterClass.DescribeChain) {
-        upstreamStatus.update(conf)
-        capabilities = RemoteCapabilities.extract(conf)
+    override fun update(conf: BlockchainOuterClass.DescribeChain): Boolean {
+        var changed = false
+        changed = changed || upstreamStatus.update(conf)
+        val newCapabilities = RemoteCapabilities.extract(conf)
+        changed = changed || (newCapabilities != capabilities)
         conf.status?.let { status -> onStatus(status) }
+        return changed
```

Comment:
And here

Reply:
Or even make `GrpcUpstream` an abstract class and move this logic in  `update` there as it the same in all places